File: App/Sources/LocWarpCore/Paths.swift
import Foundation

/// Standard file-system locations used by both the Mac app and the daemon.
public enum LocWarpPaths {
    public static var appSupportDir: URL {
        let url = FileManager.default
            .urls(for: .applicationSupportDirectory, in: .userDomainMask)[0]
            .appending(path: "LocWarp.Mac", directoryHint: .isDirectory)
        try? FileManager.default.createDirectory(at: url, withIntermediateDirectories: true)
        return url
    }

    public static var cacheDir: URL {
        let url = FileManager.default
            .urls(for: .cachesDirectory, in: .userDomainMask)[0]
            .appending(path: "LocWarp.Mac", directoryHint: .isDirectory)
        try? FileManager.default.createDirectory(at: url, withIntermediateDirectories: true)
        return url
    }

    public static var logsDir: URL {
        let home = FileManager.default.homeDirectoryForCurrentUser
        let url = home
            .appending(path: "Library", directoryHint: .isDirectory)
            .appending(path: "Logs", directoryHint: .isDirectory)
            .appending(path: "LocWarp.Mac", directoryHint: .isDirectory)
        try? FileManager.default.createDirectory(at: url, withIntermediateDirectories: true)
        return url
    }

    public static var socketPath: String {
        appSupportDir.appending(path: "locwarp.sock").path(percentEncoded: false)
    }
}

File: Daemon/locwarpd/paths.py
"""Resolve standard file-system paths used by the daemon."""

from __future__ import annotations

import os
from pathlib import Path


def app_support_dir() -> Path:
    """~/Library/Application Support/LocWarp.Mac/"""
    base = Path(os.path.expanduser("~/Library/Application Support/LocWarp.Mac"))
    base.mkdir(parents=True, exist_ok=True)
    return base


def cache_dir() -> Path:
    """~/Library/Caches/LocWarp.Mac/"""
    base = Path(os.path.expanduser("~/Library/Caches/LocWarp.Mac"))
    base.mkdir(parents=True, exist_ok=True)
    return base


def logs_dir() -> Path:
    """~/Library/Logs/LocWarp.Mac/"""
    base = Path(os.path.expanduser("~/Library/Logs/LocWarp.Mac"))
    base.mkdir(parents=True, exist_ok=True)
    return base


def socket_path() -> str:
    return str(app_support_dir() / "locwarp.sock")
